File: actor-bench/actor_bench.js
class Actor {
  constructor() {
    this.x = 0;
    this.y = 0;
    this.vx = 0;
    this.vy = 0;
  }

  update() {
    this.x += this.vx;
    this.y += this.vy;
  }
}

const createActors = num => {
  const actors = [];
  for (let i = 0; i < num; i++) {
    var newActor = new Actor();
    newActor.x = i / 10.0;
    newActor.y = (i * 2) / 10.0;
    newActor.vx = i / 100.0;
    newActor.vy = (i * 2) / 100.0;
    actors.push(newActor);
  }
  return actors;
};

const updateAll = actors => {
  for (i = 0; i < actors.length; i++) {
    const actor = actors[i];
    actor.update();
  }
};

const main = () => {
  if (process.argv.length <= 2) {
    return 1;
  }
  const start = new Date().getTime();

  const actors = createActors(10000);

  const num = parseInt(process.argv[2], 10);
  for (j = 0; j < num; j++) {
    updateAll(actors);
  }

  console.info(new Date().getTime() - start);

  console.info(actors[5000].x);
  console.info(actors[5000].y);
};

main();

File: actor-bench/actor_bench.rb
class Actor
    attr_accessor :x
    attr_accessor :y
    attr_accessor :vx
    attr_accessor :vy

    def update
      @x += @vx
      @y += @vy
    end
end

def create_actors(num)
  instances = []
  num.times {|i|
    new_actor = Actor.new
    new_actor.x = i/10.0
    new_actor.y = i*2/10.0
    new_actor.vx = i/100.0
    new_actor.vy = i*2/100.0
    instances.push(new_actor)
  }
  return instances
end

def update_all(actors)
  actors.each {|actor|
    actor.update()
  }
end


actors = create_actors(10000)
10000.times {
  update_all(actors)
}

puts actors[5000].x
puts actors[5000].y
